contract/scripts/deploy-contract.js

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env node
+/* global process, fetch, setTimeout */
+// @ts-check
+import '@endo/init';
+import fsp from 'node:fs/promises';
+import { execFile, execFileSync } from 'node:child_process';
+import { basename } from 'node:path';
+
+import { makeNodeBundleCache } from '@endo/bundle-source/cache.js';
+import { makeE2ETools } from '../tools/e2e-tools.js';
+import { NonNullish } from '../src/objectTools.js';
+
+const opt0 = {
+  service: 'agd',
+  workdir: '/workspace/contract',
+};
+
+const Usage = `
+deploy-contract [options] [--install <contract>] [--evals <proposal>...]
+
+Options:
+  --help               print usage
+  --install            entry module of contract to install
+  --evals              entry modules of core evals to run
+                       (cf rollup.config.mjs)
+  --service SVC        docker compose service to run agd (default: ${opt0.service})
+                       use . to run agd outside docker.
+  --workdir DIR        workdir for docker service (default: ${opt0.workdir})
+`;
+
+/**
+ * @param {string[]} argv
+ * @param {{ [k: string]: boolean | [] | undefined }} [style]
```

Comment:
I think the `undefined` is implied by the optionality. What is the array of?
```suggestion
 * @param {{ [k: string]: boolean | unknown[] }} [style]
```